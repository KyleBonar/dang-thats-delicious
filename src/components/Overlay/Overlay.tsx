import * as React from "react";
import styled from "styled-components";

export interface IOverlayProps {
  children: any;
  className?: string;
  enabled?: boolean;
}

class Overlay extends React.PureComponent<IOverlayProps, any> {
  public static defaultProps = {
    enabled: true
  };
  public render() {
    return (
      <div className={this.props.className}>
        <div id="overlay" />
        {this.props.children}
      </div>
    );
  }
}

const StyledOverlay = styled(Overlay)`
  position: relative;

  > div:first-child {
    position: absolute;
    width: ${props => (props.enabled ? "0" : "100%")};
    height: ${props => (props.enabled ? "0" : "100%")};
    z-index: ${props => (props.enabled ? "0" : "999")};
    background-color: ${props =>
      props.enabled ? "transparent" : "rgba(255,255,255,0.6)"};
  }
`;

export { StyledOverlay as Overlay };
