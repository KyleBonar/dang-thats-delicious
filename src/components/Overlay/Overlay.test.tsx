import * as React from "react";
import * as enzyme from "enzyme";

import { IOverlayProps, Overlay } from "./Overlay";
import {
  fakeJSXElement,
  casual,
  ITERATION_SIZE
} from "../../utils/testUtils/testUtils";

export const mockOverlay = (): IOverlayProps => ({
  children: fakeJSXElement(),
  className: casual.string,
  enabled: casual.random_element([undefined, casual.boolean])
});

const mockOverlays = new Array(ITERATION_SIZE).fill(null).map(mockOverlay);

describe("<Overlay />", () => {
  it("renders", () => {
    mockOverlays.forEach((mockOverlay: IOverlayProps) => {
      const wrapper = enzyme.shallow(
        <Overlay {...mockOverlay}>{mockOverlay.children}</Overlay>
      );

      expect(wrapper).toBeTruthy();
    });
  });

  it("matches snapshot", () => {
    mockOverlays.forEach((mockOverlay: IOverlayProps) => {
      const wrapper = enzyme.shallow(
        <Overlay {...mockOverlay}>{mockOverlay.children}</Overlay>
      );

      expect(wrapper).toMatchSnapshot();
    });
  });

  it("adds classes to parent", () => {
    mockOverlays.forEach((mockOverlay: IOverlayProps) => {
      const wrapper = enzyme.shallow(
        <Overlay {...mockOverlay}>{mockOverlay.children}</Overlay>
      );

      expect(wrapper.props().className).toContain(mockOverlay.className);
    });
  });

  it("contains expected children", () => {
    const tmpChildren = (
      <>
        <div>howdy</div>
        <span>hi</span>
      </>
    );
    const wrapper = enzyme.shallow(<Overlay>{tmpChildren}</Overlay>);

    expect(wrapper.props().children).toEqual(tmpChildren);
  });
});
