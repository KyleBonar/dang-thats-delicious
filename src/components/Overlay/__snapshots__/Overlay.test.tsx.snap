// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overlay /> matches snapshot 1`] = `
<Overlay
  className="sc-AxjAm lftuZY sint quia sed exercitationem omnis laborum exercitationem"
  enabled={false}
>
  <span
    key="4c3f2368-2393-4e09-8492-ed03e808d9d8"
  >
    dicta perferendis dolorum cupiditate illum saepe ut
  </span>
</Overlay>
`;

exports[`<Overlay /> matches snapshot 2`] = `
<Overlay
  className="sc-AxjAm lftuZY et doloremque voluptate in sunt occaecati nulla"
>
  <div
    key="68c1398f-9e79-4c41-ba45-15a740f67ee1"
  >
    dolores ipsam laboriosam non itaque voluptas ea
  </div>
</Overlay>
`;

exports[`<Overlay /> matches snapshot 3`] = `
<Overlay
  className="sc-AxjAm lftuZY saepe assumenda deserunt consequatur itaque et in"
>
  <div
    key="ded82d4e-49ed-4fec-906f-5f733a072edb"
  >
    dignissimos dolore qui voluptatem cumque nihil iste
  </div>
</Overlay>
`;

exports[`<Overlay /> matches snapshot 4`] = `
<Overlay
  className="sc-AxjAm lftuZY molestiae cupiditate autem consequatur consequatur accusantium molestiae"
  enabled={false}
>
  <div
    key="fa1039c0-0e13-457d-a1ca-ea1cf4cc8df5"
  >
    est esse quos minus deserunt nihil quasi
  </div>
</Overlay>
`;

exports[`<Overlay /> matches snapshot 5`] = `
<Overlay
  className="sc-AxjAm lftuZY perferendis nostrum aliquid ea animi architecto accusantium"
>
  <div
    key="001ad558-1c98-454a-b468-e9abf9796f71"
  >
    voluptate voluptatem est eius dolorem amet quia
  </div>
</Overlay>
`;

exports[`<Overlay /> matches snapshot 6`] = `
<Overlay
  className="sc-AxjAm jXVRFe ea totam aut aut sunt enim rem"
  enabled={true}
>
  <span
    key="fc56811d-a6af-4c68-9078-f1b61de77e61"
  >
    culpa suscipit totam libero necessitatibus quis ratione
  </span>
</Overlay>
`;

exports[`<Overlay /> matches snapshot 7`] = `
<Overlay
  className="sc-AxjAm lftuZY quo aut quia cumque architecto eos quos"
  enabled={false}
>
  <span
    key="745aff4d-4dbe-4062-bc76-6c0922d664e8"
  >
    omnis qui illo excepturi esse porro impedit
  </span>
</Overlay>
`;

exports[`<Overlay /> matches snapshot 8`] = `
<Overlay
  className="sc-AxjAm lftuZY et placeat ipsum assumenda officia aut id"
  enabled={false}
>
  <span
    key="03cd632b-ab80-4725-92ba-db7ec3d2fc86"
  >
    delectus qui qui ab officiis omnis rem
  </span>
</Overlay>
`;
